lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2991,19 +2991,19 @@ class PrimalGenCloner final
   /// Primal has qualified ownership. We assign store ownership qualifier while
   /// cloning the `store` instruction.
   void visitStoreInst(StoreInst *si) {
-    auto destTy = si->getDest()->getType().getASTType();
+    auto destTy = getOpType(si->getDest()->getType());
     auto loc = remapLocation(si->getLoc());
-    auto soq = getBufferSOQ(getOpASTType(destTy), *getPrimal());
+    auto soq = getBufferSOQ(destTy.getASTType(), *getPrimal());
     getBuilder().createStore(loc, getOpValue(si->getSrc()),
                              getOpValue(si->getDest()), soq);
   }
 
   /// Primal has qualified ownership. We assign load ownership qualified while
   /// cloning the `load` instruction.
   void visitLoadInst(LoadInst *li) {
```

Comment:
These two visitors should ideally be deleted. Did you try that?

Reply:
I will try this later today.

Reply:
Discussed offline. I'm gonna experiment with removing these two visitors and merge it to make progress.

I wrote these two visitors initially because there was some cloning issues regarding the ownership qualifier. Not sure if that situation changed after these rounds of merging.

Reply:
It did work.